src/factories/plugin-factory.ts

```diff
@@ -63,6 +65,7 @@ const pluginFactories: Record<string, PluginBuilder> = {
       {
         ...options,
         ...(options.type as WorkspacePluginOptions),
+        merge: determineMerge(options),
```

Comment:
This will override the `merge` option provided by the user (`WorkspacePluginOptions` has an optional `merge`).

Also, it seems weird for `determineMerge` to be checking to see which type of plugin is being configured. `determineMerge` is already called in a block chosen based on the plugin type. We should just explicitly set the default in the plugin factory for each type.